test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Host.Tests.ps1

```diff
@@ -1,18 +1,93 @@
-﻿Describe "Write-Host DRT Unit Tests" -Tags "CI" {
+﻿Describe "Write-Host DRT Unit Tests" -Tags "Slow","Feature" {
 
     BeforeAll {
         $powershell = Join-Path -Path $PsHome -ChildPath "powershell"
+
+        $testData = @(
+            @{ Name = '-Separator';Command = "Write-Host a,b,c -Separator '+'"; returnCount = 1; returnValue = @("a+b+c") }
+            @{ Name = '-NoNewline=true';Command = "Write-Host a,b -NoNewline:`$true;Write-Host a,b"; returnCount = 1; returnValue = @("a ba b") }
+            @{ Name = '-NoNewline=false';Command = "Write-Host a1,b1;Write-Host a2,b2"; returnCount = 2; returnValue = @("a1 b1","a2 b2") }
+        )
+    }
+
+    It "write-Host works with '<Name>' switch" -TestCases:$testData -Pending:$IsOSX {
+        param($Command, $returnCount, $returnValue)
+
+        [array]$result = & $powershell -noprofile $Command
+        $result.Count | Should Be $returnCount
+        $result[0] | Should Be $returnValue[0]
+        $result[1] | Should Be $returnValue[1]
     }
+}
+
+Describe "Write-Host test wrong colors" -Tags "CI" {
 
+    BeforeAll {
+        $testWrongColor = @(
+            @{ ForegroundColor = -1; BackgroundColor = 'Red' }
+            @{ ForegroundColor = 16; BackgroundColor = 'Red' }
+            @{ ForegroundColor = 'Red'; BackgroundColor = -1 }
+            @{ ForegroundColor = 'Red'; BackgroundColor = 16 }
+        )
+    }
+
+    It 'Should throw if color is invalid: ForegroundColor = <ForegroundColor>; BackgroundColor = <BackgroundColor>' -TestCases:$testWrongColor {
+      param($ForegroundColor, $BackgroundColor)
+      try
+      {
+          Write-Host "No output from Write-Host" -ForegroundColor $ForegroundColor -BackgroundColor $BackgroundColor
+          throw "No Exception!"
+      }
+      catch { $_.FullyQualifiedErrorId | Should Be 'CannotConvertArgumentNoMessage,Microsoft.PowerShell.Commands.WriteHostCommand' }
+    }
+}
+
+Describe "Write-Host test with TestHostCS" -Tags "CI" {
+
+    BeforeAll {
+        if ( -not (get-module -ea SilentlyContinue TestHostCS )) {
+            $hostmodule = Join-Path $PSScriptRoot "../../Common/TestHostCS.psm1"
+            import-module $hostmodule
+        }
+        $th = New-TestHost
+        $rs = [runspacefactory]::Createrunspace($th)
+        $rs.open()
+        $ps = [powershell]::Create()
+        $ps.Runspace = $rs
+
+        $testHostCSData = @(
+            @{ Name = 'defaults';Command = "Write-Host a,b,c"; returnCount = 1; returnValue = @("White:Black:a b c:NewLine") }
```

Comment:
I think this would be easier to read and review if each key/value pair was on a line by itself.